syncto/tests/test_functional.py

```diff
@@ -293,3 +286,33 @@ def test_put_return_a_404_in_case_of_unknown_resource(self):
             response=response)
         self.app.put_json(RECORD_URL, RECORD_EXAMPLE,
                           headers=self.headers, status=404)
+
+
+class WriteSafeguardTest(BaseViewTest):
+    def get_app_settings(self, extra=None):
+        settings = super(WriteSafeguardTest, self).get_app_settings(extra)
+        settings['syncto.record_tabs_put_enabled'] = False
+        settings['syncto.record_tabs_delete_enabled'] = False
+        return settings
+
+    def test_record_put_is_disabled_on_meta_and_crypto(self):
+        url = RECORD_URL.replace('tabs', 'meta')
+        self.app.put_json(url, RECORD_EXAMPLE,
+                          headers=self.headers, status=405)
+
+    def test_record_delete_is_disabled_on_meta_and_crypto(self):
+        url = RECORD_URL.replace('tabs', 'meta')
+        self.app.delete(url, headers=self.headers, status=405)
+
+    def test_record_put_is_disabled(self):
+        self.app.put_json(RECORD_URL, RECORD_EXAMPLE,
+                          headers=self.headers, status=405)
+
+    def test_record_delete_is_disabled(self):
+        self.app.delete(RECORD_URL, headers=self.headers, status=405)
+
+    def test_message_provides_details(self):
+        r = self.app.put_json(RECORD_URL, RECORD_EXAMPLE,
+                              headers=self.headers, status=405)
+        expected = 'Endpoint disabled for this collection in configuration.'
```

Comment:
Please use the [FormattedErrorMixin](https://github.com/mozilla-services/cliquet/blob/master/cliquet/tests/test_views_errors.py#L53-L55)